### ISSUE TITLE
routing using enum and parameter not working on production build

### ISSUE THREAD

**Issue by webcat12345** [angular/angular#32554]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [x] doc
```

### Is this a regression?

No, it's not something regression.

### Description

What we want to do is just a smart way to define routes and make it easy to maintenance, for example, changing the paths, and implementing on HTML with pre-defined enums or constants. All working well, but if we use this way with a parameter, it doesn't work on production mode. We are thinking this is limitation of AOT compiler to Javascript syntax, but that document doesn't give any idea about this issue. We would like to report this issue to `angular-cli` repository, guessing this is related with the compiler.

Thank you.

## 🔬 Minimal Reproduction
1. Define `enum/const`s for routing example
```
    export const
      ROOT = 'contacts',
      PEOPLE = 'people',
      OFFICES = 'offices';
```
2. Use defined values for routes
```
  {
    path: RouterMap.Contacts.PEOPLE, components: PeopleComponent
  },
  {
    path: RouterMap.Contacts.PEOPLE + '/:id', component: PersonComponent
  },
```
3. Check `/people/{person_id}` on brower (local). All works fine
4. After compiling **AOT** this routing won't work. It is maybe being translated into 
```
path: '[Object]' + '/:id'
```
5. Is related with [the AOT Limitation](https://angular.io/guide/aot-compiler#expression-syntax-limitations), (but it doesn't make sense)

## 🔥 Exception or Error

No error or exceptions, just defined routing is not working on production.

## 🌍 Your Environment
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.0.6
Node: 12.2.0
OS: linux x64
Angular: 8.0.3
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.6
@angular-devkit/build-angular     0.800.6
@angular-devkit/build-optimizer   0.800.6
@angular-devkit/build-webpack     0.800.6
@angular-devkit/core              8.0.6
@angular-devkit/schematics        8.0.6
@angular/cli                      8.0.6
@ngtools/webpack                  8.0.6
@schematics/angular               8.0.6
@schematics/update                0.800.6
rxjs                              6.4.0
typescript                        3.4.5
webpack                           4.30.0
</code></pre>

**Anything else relevant?**
I am still not sure, but I am wondering this document may require some updates.

https://angular.io/guide/aot-compiler#expression-syntax-limitations

**Comment by BruneXX** [comment]
Hi, I'm trying to find the other issue related with this

**Comment by BruneXX** [comment]
I've found it! #31227

**Comment by atscott** [comment]
Hey @BruneXX, can you try this with the latest RC version (9.0.0-rc.x) with the new Ivy compiler and runtime? It's possible that it will work with that version, but it's hard to know for sure without a reproduction.

**Comment by BruneXX** [comment]
Hi @atscott I'll try this soon, BTW I've tried this with angular 8.2.14 and it works perfect. Thanks to @sibest19 on #31227

**Comment by JoostK** [comment]
I am positive this issue has been resolved in Ivy, however from your sample code `RouterMap.Contacts.PEOPLE` it looks like you're using namespaces which is currently somewhat broken in Ivy. The issue with namespaces is fixed in pending PR #36106.

**Comment by BruneXX** [comment]
Hi @JoostK and @atscott this is currently working in Angular 9.0.6 I've recently migrated from Angular 8.2.14 and I can confirm that using routes from exported object, like:

``` javascript
const ROUTES = {
  route1: 'myroute1',
  route2: 'myroute2'
}
```

It's currently working :+1:

**Comment by JoostK** [comment]
@BruneXX Excellent, thanks for confirming!

**Comment by JoostK** [comment]
The namespace issue has just been resolved now that #36106 has landed, so I'll go ahead and close this as fixed.
